### PR TITLE
Set encoding to UTF8 on Windows and fix :exec

### PR DESCRIPTION
**Pull request by melted**
For some inexplicable reason Haskell hardcodes latin1 as the encoding on
Windows. This stops the output when an unicode character is reached. It's better
 all over to set the encoding to UTF8, if the user has a UTF8 terminal like
mintty it looks good. If the user has a terminal with latin1, there will
be some bad chars but they will see the entire text otherwise.

### CODE REVIEW

**Comment by melted** [comment]
The :exec fix was made on the utf8 fix, so they have to go in the same PR.

File: main/Main.hs [hunk 1]
```diff
@@ -38,7 +39,9 @@ import Paths_idris
 -- on with the REPL.
 
 main :: IO ()
-main = do opts <- runArgParser
+main = do
```
Comment (Heather) [inline main/Main.hs]: I don't like new line here

**Comment by Heather** [comment]
@melted doesn't encoding came from system defaults?
What if there will be linux with non-utf8 locale as default one?

**Comment by LeifW** [comment]
We've talked about just ignoring system specified encoding and just assuming .idr files are in utf8 (gotten build errors in the past when the stdlib files have unicode in them and people try builing them in a C locale or something).
Do we want to "if windows" check before `hSetEncoding stdout utf8`? Or should we just set it that way on all platforms?

**Comment by melted** [comment]
This fixes the output at the repl and the terminal that will choke with Unicode characters if the encoding is set to latin1. The specific case that bugged me was that the help text was truncated thanks to a Unicode apostrophe.

----- Ursprungligt meddelande -----
Från: "Leif Warner" <notifications@github.com>
Skickat: ‎2015-‎02-‎23 07:33
Till: "idris-lang/Idris-dev" <Idris-dev@noreply.github.com>
Kopia: "Niklas Larsson" <metaniklas@gmail.com>
Ämne: Re: [Idris-dev] Set encoding to UTF8 on Windows and fix :exec (#1937)

We've talked about just ignoring system specified encoding and just assuming .idr files are in utf8 (gotten build errors in the past when the stdlib files have unicode in them and people try builing them in a C locale or something).
Do we want to "if windows" check before hSetEncoding stdout utf8? Or should we just set it that way on all platforms?

**Comment by Heather** [comment]
Error msg:DeleteFile "test.exe": does not exist (╨Э╨╡ ╤Г
╨┤╨░╨╡╤В╤Б╤П ╨╜╨░╨╣╤В╨╕ ╤Г╨║╨░╨╖╨░╨╜╨╜╤Л╨╣ ╤Д╨░╨╣╨╗.)

I'm so happy with UTF8 :dancer:

**Pull request merged**